foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/DataLineageAiopsReferencesMigration.java

```diff
@@ -0,0 +1,85 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.boozallen.aissemble.upgrade.util.DataLineageUtil.*;
+import static com.boozallen.aissemble.upgrade.util.FileUtils.replaceInFile;
+
+public class DataLineageAiopsReferencesMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(DataLineageAiopsReferencesMigration.class);
+
+    protected String getAissembleVersion() {
+        return getMavenProject().getProperties().getProperty("version.aissemble");
+    }
+
+    /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        try (BufferedReader dataLineageConfig = new BufferedReader((new FileReader(file)))) {
+
+            String line;
+            while ((line = dataLineageConfig.readLine()) != null) {
+                Pattern pattern = Pattern.compile(VERSION_AISSEMBLE_PRE_1_7_0_REGEX);
+                Matcher matcher = pattern.matcher(line);
+
+                if (matcher.find()) {
+                    shouldExecute = true;
```

Comment:
A: Short-circuit so we stop reading the file as soon as we identify it needs to be migrated. Either with a direct `return` here or (my preference) adding a `&& !shouldExecute` to the `while` condition.

Reply:
Made this change, thanks for the recommendation! :)